### DISCUSSION TITLE
How does this equation get its weak form？

### DISCUSSION THREAD

**Discussion by zong-yanyun** [idaholab/moose#23110] [Q&A Getting Started] [answered]
∂c/∂t=∇[D(∇c+c⋅Q/(RT^2 )⋅∇T)  , c is to be evaluated , T is temperature , t is time .The other parameters is about material. I'm not sure whether this equation can be gotten its weak form, whether it can be solved by moose.

**Comment by WilkAndy** [comment, marked as answer]
Assuming that your equation is missing a square-bracket at the end (just before the comma), the weak form of the right-hand side is just performing integration by parts (taking the first gradient operator and applying it to the test function).  Your physics is modelled in MOOSE using:
- https://mooseframework.inl.gov/source/kernels/TimeDerivative.html
- https://mooseframework.inl.gov/source/kernels/MatDiffusion.html with diffusivity given by "D" which is a Material in the input file (I assume that's what you mean by "The other parameters is about material")
- https://mooseframework.inl.gov/source/kernels/ConservativeAdvection.html , with advective velocity given by D Q grad(T) / R / T^2.   You'll have to make an AuxVariable for this advective velocity - eg, you could use 3 https://mooseframework.inl.gov/source/auxkernels/ParsedAux.html (one for each velocity component).

**Reply by zong-yanyun** [reply]
Thank you!

**Reply by hugary1995** [reply]
Here is a complete example on deriving weak form:
https://mooseframework.inl.gov/getting_started/examples_and_tutorials/tutorial01_app_development/step04_weak_form.html

**Reply by zong-yanyun** [reply]
Thank you for your help. I have got the weak form, but there is ∇T (T is not the  parameter I want to solve) in the weak form. How should I define the temperature T? Is it still in material？

**Reply by zong-yanyun** [reply]
One more thing,  Q and c are related (c is to be evaluated) , how should I define the Q  in moose?

**Reply by GiudGiud** [reply]
If T is not to be solved for, if it's a known quantity, then T can be an auxiliary variable and initialized using a FunctionIC. 
A function in MOOSE is a known quantity in time & space. 

It could also be a material property if it depends on local quantities, especially if those local quantities are being solved for. 
Q should likely be a material property
